node/modules/lp2p/rcmgr.go

```diff
@@ -27,9 +27,10 @@ import (
 
 func ResourceManager(connMgrHi uint) func(lc fx.Lifecycle, repo repo.LockedRepo) (network.ResourceManager, error) {
 	return func(lc fx.Lifecycle, repo repo.LockedRepo) (network.ResourceManager, error) {
+		isFullNode := repo.RepoType().Type() == "FullNode"
 		envvar := os.Getenv("LOTUS_RCMGR")
-		if envvar == "0" {
-			// this is enabled by default; specify LOTUS_RCMGR=0 to disable
+		if (isFullNode && envvar == "0") || // enable by default for full nodes
```

Comment:
Logic seems wrong, I think it should be:
```suggestion
		if (isFullNode && envvar != "0") || // enable by default for full nodes
```

Reply:
we might want to fix the comment to avoid tripping casual reading.
Any suggestions?